### PR TITLE
Introduce a tool to display memory usage.

### LINKED ISSUE #34: spdmlib use a lot of stack. It may cause stack overflow in small stack environment.
Some old data:

```
default config:
requester stack use: 808040 bytes
responder stack use: 818632 bytes

change config: max_session_count = 10
requester stack use: 1580888 bytes
responder stack use: 1591480 bytes

change config: max_session_count = 1
requester stack use: 421640 bytes
responder stack use 447560 bytes

each session costs ~120k bytes
```

We need re-calculate latest data with `hashed-transcript-data`.

### PR DESCRIPTION
**Pull request by xiaoyuxlu**
Fix #34 #155 #151

1. minor fix
2. add max heap usage info
3. add a script to display memory usage.

### CODE REVIEW

**Comment by xiaoyuxlu** [comment]
results on 60f7c4ab6865b64fba0f2d019245641400619367
```
|                         |      Requester        |      Responder        |
|                         | stack     |  heap     | stack     |  heap     |
| ----------------------- |-----------|-----------|-----------|-----------|
| async + release         |     412250|     321231|     337994|     254263|
| tokio + release         |     412682|    1134431|     338346|    1067463|
|async + releas + w/o hash|     525530|     369919|     446234|     303143|
| sync + release          |     480802|      68655|     151722|      72647|
|sync + release + w/o hash|     611130|      68463|     247354|      72647|
| async + debug           |     412314|     321231|     333930|     254263|
| sync + debug            |     480834|      68655|     147610|      72647|
```

File: sh_script/collect_memory_usage.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            "cargo run --release -p spdm-responder-emu --no-default-features --features=spdm-ring,hashed-transcript-data,async-tokio,test_stack_size,test_heap_size"
+        ),
+        (
+            "async + releas + w/o hash",
+            "cargo run --release -p spdm-requester-emu --no-default-features --features=spdm-ring,async-executor,test_stack_size,test_heap_size",
+            "cargo run --release -p spdm-responder-emu --no-default-features --features=spdm-ring,async-executor,test_stack_size,test_heap_size"
+        ),
+        (
+            " sync + release          ",
+            "cargo run --release -p spdm-requester-emu --no-default-features --features=spdm-ring,hashed-transcript-data,async-executor,test_stack_size,test_heap_size,is_sync",
```
Comment (jyao1) [inline sh_script/collect_memory_usage.py]: why `async-executor` is added?

File: sh_script/collect_memory_usage.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+        (
+            " async + debug           ",
+            "cargo run -p spdm-requester-emu --no-default-features --features=spdm-ring,hashed-transcript-data,async-executor,test_stack_size,test_heap_size",
+            "cargo run -p spdm-responder-emu --no-default-features --features=spdm-ring,hashed-transcript-data,async-executor,test_stack_size,test_heap_size"
+        ),
+
+        (
+            " sync + debug            ",
+            "cargo run -p spdm-requester-emu --no-default-features --features=spdm-ring,hashed-transcript-data,async-executor,test_stack_size,test_heap_size,is_sync",
```
Comment (jyao1) [inline sh_script/collect_memory_usage.py]: why async-executor?

File: sh_script/collect_memory_usage.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    max_heap_usage_requester = parse_number(
+        out_requester, r"max heap usage: (\d+)")
+
+    return (name, max_stack_usage_requester, max_heap_usage_requester, max_stack_usage_responder, max_heap_usage_responder)
+
+
+def main():
+    test_vector = [
+        (
+            " async + release         ",
```
Comment (jyao1) [inline sh_script/collect_memory_usage.py]: please use "async-executor"

File: sh_script/collect_memory_usage.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            "cargo run --release -p spdm-requester-emu --no-default-features --features=spdm-ring,hashed-transcript-data,async-executor,test_stack_size,test_heap_size",
+            "cargo run --release -p spdm-responder-emu --no-default-features --features=spdm-ring,hashed-transcript-data,async-executor,test_stack_size,test_heap_size"
+        ),
+        (
+            " tokio + release         ",
```
Comment (jyao1) [inline sh_script/collect_memory_usage.py]: please use "async-tokio"

File: sh_script/collect_memory_usage.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            "cargo run --release -p spdm-requester-emu --no-default-features --features=spdm-ring,hashed-transcript-data,async-tokio,test_stack_size,test_heap_size",
+            "cargo run --release -p spdm-responder-emu --no-default-features --features=spdm-ring,hashed-transcript-data,async-tokio,test_stack_size,test_heap_size"
+        ),
+        (
+            "async + releas + w/o hash",
```
Comment (jyao1) [inline sh_script/collect_memory_usage.py]: use "raw transcript"

File: test/spdm-requester-emu/src/main.rs [hunk 6]
```diff
@@ -1283,7 +1283,7 @@ fn new_logger_from_env() -> SimpleLogger {
 fn emu_main() {
     td_benchmark::StackProfiling::init(
         0x5aa5_5aa5_5aa5_5aa5,
-        EMU_STACK_SIZE - 0x40000, // main function stack
+        EMU_STACK_SIZE - 0x60000, // main function stack
```
Comment (jyao1) [inline test/spdm-requester-emu/src/main.rs]: why 0x60000?

**Review by jyao1 — APPROVED** [review]

**Pull request merged**